models/clm/clm_to_dart.rst

```diff
@@ -0,0 +1,198 @@
+      ncdump -f F -v T_SOISNO clm_restart.nc > T_SOISNO_org.txt
+      ./clm_to_dart
+      ncdump -f F -v T_SOISNO clm_restart.nc > T_SOISNO_new.txt
+      diff T_SOISNO_org.txt T_SOISNO_new.txt
+
+
+Namelist
+--------
+
+Namelists start with an ampersand '&' and terminate with a slash '/'. 
+Character strings that contain a '/' must be enclosed in quotes to prevent 
+them from prematurely terminating the namelist. These are the defaults:
+
+::
+
+   &clm_to_dart_nml
+     clm_restart_file  = 'clm_restart.nc'
+     verbose           = 0
+    /
+
+
+.. container::
+
+   ================== ==================== ================================================================= 
+   Item               Type                 Description                                                     
+   ================== ==================== ================================================================= 
+   clm_restart_file   character(len=256)   Path name of the CLM restart file to be overwritten.
+   verbose            integer              | Flag to control how much run-time output is created.
+                                           | 0   is very little output.
+                                           | 1   reports which variables are being updated.
+                                           | 2   reports all 2d variables with columns as one dimension.
+                                           | 3   reports the variables and columns that have traces of snow.
+                                           | *Warning*, 3 can generate a lot of output.
+   ================== ==================== =================================================================
+
+
+Modules used
+------------
+
+::
+
+   assimilation_code/modules/utilities/netcdf_utilities_mod.f90
+   assimilation_code/modules/utilities/null_mpi_utilities_mod.f90
+   assimilation_code/modules/utilities/time_manager_mod.f90
+   assimilation_code/modules/utilities/types_mod.f90
+   assimilation_code/modules/utilities/utilities_mod.f90
+
+
+Files
+-----
+
+-  clm_restart_file, specified by namelist;
+-  namelistfile; ``input.nml``
+
+
+Discussion of Indeterminate Values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To explore the most robust way to replace the indeterminate values with the 
+DART-required *_FillValue*, a low-resolution run (10 deg x 15 deg) was performed 
+using the **f10_f10_musgs** tag on the **release-clm5.0.34** branch.
+The ``clm_to_dart_nml`` verbosity was set to 3 and the run-time output was captured 
+and is summarized below. The output has been pruned to improve clarity and the 
+column headers have been added. Only the output for the *T_SOISNO* variable is 
+shown as it clearly illustrates the situation.
+
+The **frac_sno** variable (snow cover fraction) is used to determine when
+there is a reliable trace amount of snow and the original values in the snow layer
+closest to the ground are left alone. This was determined to be a more useful
+indicator than **H2OSNO**.  
+
```

Comment:
In other words all values of T_SOISNO that do not have  frac_sno >0 are changed to _FillValue before DART operates on the file.   Then the original indeterminate value is reinserted back into the file in dart_to_clm